interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,200 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const mediaHubPermanentSlotAppArmor = `
+# Description: Allow operating as the the media-hub service.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=session
+    name="core.ubuntu.media.Service",
+
+dbus (receive, send)
+    bus=session
+    path=/com/ubuntu/media/Service{,/**}
+    interface=org.freedesktop.DBus{,.*}
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=unconfined),
+`
+
+const mediaHubConnectedSlotAppArmor = `
+# Allow connected clients to interact with the player
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to manage Player sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow querying AppArmor
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionAppArmorSecurityContext"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const mediaHubConnectedPlugAppArmor = `
+# Description: Allow using media-hub service.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
This is a very open rule, but the ConnectedSlot rules are more fine-grained. If it is safe for the client to talk to the service on all methods, feel free to adjust the ConnectedSlot policy to use:
```
dbus (receive, send)
￼    bus=session
    path=/core/ubuntu/media/Service{,/**}
    peer=(label=###PLUG_SECURITY_TAGS###),
```

If there are some APIs that should not be accessible to plugs, please list the safe ones in both ConnectedSlot and ConnectedPlug.